internal/rpcserver/rpcwebsocket.go

```diff
@@ -110,28 +110,28 @@ func (s *rpcServer) WebsocketHandler(conn *websocket.Conn, remoteAddr string, au
 	// disconnected), remove it and any notifications it registered for.
 	client, err := newWebsocketClient(s, conn, remoteAddr, authenticated, isAdmin)
 	if err != nil {
-		rpcsLog.Errorf("Failed to serve client %s: %v", remoteAddr, err)
+		log.Errorf("Failed to serve client %s: %v", remoteAddr, err)
 		conn.Close()
 		return
 	}
 	s.ntfnMgr.AddClient(client)
 	client.Start()
 	client.WaitForShutdown()
 	s.ntfnMgr.RemoveClient(client)
-	rpcsLog.Infof("Disconnected websocket client %s", remoteAddr)
+	log.Infof("Disconnected websocket client %s", remoteAddr)
 }
 
-// wsNotificationManager is a connection and notification manager used for
+// WsNotificationManager is a connection and notification manager used for
 // websockets.  It allows websocket clients to register for notifications they
 // are interested in.  When an event happens elsewhere in the code such as
 // transactions being added to the memory pool or block connects/disconnects,
 // the notification manager is provided with the relevant details needed to
 // figure out which websocket clients need to be notified based on what they
 // have registered for and notifies them accordingly.  It is also used to keep
 // track of all connected websocket clients.
-type wsNotificationManager struct {
+type WsNotificationManager struct {
```

Comment:
Would do you think about adding delegates for the websocket notification functions to the `rpcserver.Server` (it would be `RPCServer` in the commit order since it would take the place of the the `rpcserver: Export wsNotificationManager` commit) instead of exporting this?

I suggest this for a few reasons:
- It would be significantly less code changes
- I think it would be more consistent with `NotifyNewTransactions` which is exported from the rpc server
- From a caller perspective, it's a more unified API
- It seems to me that this is internal machinery that is better left unexported

e.g.

```Go
func (s *RPCServer) NotifyWork(templateNtfn *mining.TemplateNtfn) {
	s.ntfnMgr.NotifyWork(templateNtfn)
}
```

Then over where it's called `r.NtfnMgr().NotifyWork(templateNtfn)` just becomes `r.NotifyWork(templateNtfn)`.

The same deal for the other notification funcs.

Reply:
I did consider that as well but saw that there were more than a couple of those functions (looks like 8 or so of the `NotifyXYZ` functions) which swayed me to go the other route.  But I do agree with your points, especially that it results in less being exported from rpcserver, so I'll make that update

Reply:
Nice.  Thanks.  I do think it looks better like this and, as I suspected, it appears that a bit less code had to be changed.